driver-core/src/main/com/mongodb/internal/async/client/ClientSessionHelper.java

```diff
@@ -61,6 +61,8 @@ void createClientSession(final ClientSessionOptions options, final OperationExec
                 && clusterDescription.getLogicalSessionTimeoutMinutes() != null
                 && clusterDescription.getType() != ClusterType.STANDALONE) {
             callback.onResult(createClientSession(options, executor), null);
+        } else if (mongoClient.getCluster().isClosed()) {
+            callback.onResult(null, null);
```

Comment:
This is a good catch, but what happens if both are returned as null?  What error will the user see?

Reply:
The user would not see an error in this case. The failure would be silent, and the client would have nothing to process. But that does not inform the user of any underlying failures. I have added a `MongoException` here, similar to the errors added in JAVA-3487,  to indicate the cluster is closed. Since I have been unable to reproduce the errors in the ticket locally, I suspect we will see a ticket in the future with this exception.